Tests fail with PyPy3 because of different return values for iter()

Hi,
It looks like PyPy3 doesn't use the same names for types returned by `iter()`.

The following code:
```py
print(iter([1,2,3]))
print(iter((1, 2)))
print(iter(range(10)))
```
will return this with Python 3.8:
```
<list_iterator object at 0x7f711541c040>
<str_iterator object at 0x7f711541c040>
<range_iterator object at 0x7f71153db2d0>
```
and this with PyPy3 7.3.0:
```
<sequenceiterator object at 0x00007f2a1a2cc8d8>
<sequenceiterator object at 0x00007f2a1a2d45a0>
<rangeiterator object at 0x00007f2a1a2c1a10>
```

This makes the package's tests fail:
```
=================================== FAILURES ===================================
__________________ [doctest] jaraco.itertools.always_iterable __________________
881 
882     Given an object, always return an iterable. If the item is not
883     already iterable, return a tuple containing only the item. If item is
884     None, an empty iterable is returned.
885 
886     >>> always_iterable([1,2,3])
Expected:
    <list_iterator...>
Got:
    <sequenceiterator object at 0x00007f51eaac4b60>

/tmp/jaraco.itertools/jaraco/itertools.py:886: DocTestFailure
881 
882     Given an object, always return an iterable. If the item is not
883     already iterable, return a tuple containing only the item. If item is
884     None, an empty iterable is returned.
885 
886     >>> always_iterable([1,2,3])
887     <list_iterator...>
888     >>> always_iterable('foo')
Expected:
    <tuple_iterator...>
Got:
    <sequenceiterator object at 0x00007f51ecb57400>

/tmp/jaraco.itertools/jaraco/itertools.py:888: DocTestFailure
881 
882     Given an object, always return an iterable. If the item is not
883     already iterable, return a tuple containing only the item. If item is
884     None, an empty iterable is returned.
885 
886     >>> always_iterable([1,2,3])
887     <list_iterator...>
888     >>> always_iterable('foo')
889     <tuple_iterator...>
890     >>> always_iterable(None)
Expected:
    <tuple_iterator...>
Got:
    <sequenceiterator object at 0x00007f51ecb57440>

/tmp/jaraco.itertools/jaraco/itertools.py:890: DocTestFailure
883     already iterable, return a tuple containing only the item. If item is
884     None, an empty iterable is returned.
885 
886     >>> always_iterable([1,2,3])
887     <list_iterator...>
888     >>> always_iterable('foo')
889     <tuple_iterator...>
890     >>> always_iterable(None)
891     <tuple_iterator...>
892     >>> always_iterable(range(10))
Expected:
    <range_iterator...>
Got:
    <rangeiterator object at 0x00007f51ed2a64f8>

/tmp/jaraco.itertools/jaraco/itertools.py:892: DocTestFailure
===================== 1 failed, 34 passed in 0.26 seconds ======================
```

Is this something you could work around or do I need to open an upstream PyPy issue?

I've looked into skipping the doctests and it doesn't look pretty.

--- a/conftest.py
+++ b/conftest.py
@@ -1,0 +1,12 @@
+import sys
+
+
+def pytest_collection_modifyitems(session, config, items):
+    remove_broken_tests(items)
+
+
+def remove_broken_tests(items):
+    # Remove broken tests for PyPy3
+    if hasattr(sys, 'pypy_version_info'):
+        broken_test_names = ['jaraco.itertools.always_iterable']
+        items[:] = (item for item in items if item.name not in broken_test_names)

